tests/test_interpreter.py

```diff
@@ -167,3 +183,34 @@ def test_find_python_interpreter_missing_on_path_raises(monkeypatch):
         find_python_interpreter(interpreter)
         assert "Python Launcher" in str(e)
         assert "on your PATH" in str(e)
+
+
+@pytest.mark.skipif(sys.platform.startswith("win"), reason="Looks for python3")
+def test_fetch_missing_python(monkeypatch):
+    major, minor = _derive_target_python_version()
+
+    python_path = find_python_interpreter(f"{major}.{minor}", fetch_missing_python=True)
+    assert python_path is not None
```

Comment:
I'd probably ensure the Python really isn't found (e. g. for users who simply run `pytest`):

```suggestion
   python_version = f"{major}.{minor}"
   
   with pytest.raises(InterpreterResolutionError):
       find_python_interpreter(python_version)

    python_path = find_python_interpreter(python_version, fetch_missing_python=True)
    assert python_path is not None
```

Let's add a variable `python_version` as demonstrated here, BTW, to simplify.

Reply:
Indenting with GH code suggestions is a pain.... This wasn't really meant to be committed as-is, sorry. The other test should also be amended, and the variable can be used once more.

Reply:
gotcha. I'll revert and commit